tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -424,6 +436,50 @@ public:
       }
    };
 
+   // clang-format off
+   /**
+   \class ROOT::Experimental::RNTupleDescriptor::RClusterDescriptorRange
+   \ingroup NTuple
+   \brief Used to loop over all the clusters of an ntuple (in unspecified order)
+
+   Enumerate all cluster IDs from the cluster descriptor.  No specific order can be assumed, use
+   FindNextClusterId and FindPrevClusterId to travers clusters by entry number.
+   */
+   // clang-format on
+   class RClusterDescriptorRange {
```

Comment:
Do you mean the suffix 'Range'?  If so would you have some sort of range start and end end that is different in each instance of `RClusterDescriptorRange` ?  (ie. see `std::span` or even `RNTupleClusterRange` :) )

Maybe `RClusterDescriptorIteratable` and then instead of `GetClusterRange` something like `make_iteratable` or `GetClusterIteratable`?  (Another alternative I can think of is to use 'Collection` instead of `Iteratable` but that might be over-promising)

Reply:
As discussed, I added a corresponding TODO because there are other `*Range` classes similar to `RClusterDescriptorRange` that should be renamed consistently.